[llvm-lit][test] Resolved typo in raising InternalShellError for export command in lit's internal shell

connieyzhu opened the pull request:
This patch fixes the incorrect usage of lit's built-in `export` command. There is a typo in raising the error itself where the error being raised had the wrong number of parameters passed in.

Fixes https://github.com/llvm/llvm-project/issues/102386.

connieyzhu committed:
[llvm-lit] Fixed typo in raising InternalShellError in built-in export
command

This patch fixes a typo in lit's built-in export command, which returns
an InternalShellError when more than one argument is given to the
export. The InternalShellError was missing the cmd parameter when
raised, leading to issues when a test using export fails in
high-address-dereference.c in compiler-rt.

connieyzhu committed:
[compiler-rt][test] Rewrote test to fix usage of export with 2 arguments

This patch fixes the incorrect usage of lit's built-in export command in
this compiler-rt test. The export command only allows for one argument,
but %env_asan_opts expands to "env ASAN_OPTIONS", which gives export two
arguments instead of only one. Instead of using export, which propogates
the environment variable value to all subsequent commands, the
environment variable is manually set before all RUN lines following the
original export line.

ilovepi commented on llvm/utils/lit/lit/TestRunner.py:
```diff
@@ -356,7 +356,7 @@ def executeBuiltinPopd(cmd, shenv):
     """executeBuiltinExport - Set an environment variable."""
     if len(cmd.args) != 2:
-        raise InternalShellError("'export' supports only one argument")
+        raise InternalShellError(cmd, "'export' supports only one argument")
```
There should be an accompanying test.

connieyzhu replied:
I created the test in my latest change. I only made a test to show the failing behavior that is used here, but I wasn't sure if I should make tests to show that the export command works. There are other lit tests that use the export command, which do show to some extent that the command works. Do you think that's enough, or should I add more tests?

ilovepi replied:
Your change literally only affects the error message, so I don't think that's required. As long as other tests are passing/failing the same as before your patch then I don't think you need to update the world. Now, it would be nice to improve the state of testing, but that shouldn't block you from making the error consistent.

ilovepi commented on compiler-rt/test/asan/TestCases/Posix/high-address-dereference.c:
```diff
@@ -5,12 +5,11 @@
-// RUN: not %run %t 0x0000000000001000 2>&1 | FileCheck %s --check-prefixes=LOW2,HINT-NONE
-// RUN: not %run %t 0x4141414141414141 2>&1 | FileCheck %s --check-prefixes=HIGH,HINT-HIGHADDR
-// RUN: not %run %t 0xFFFFFFFFFFFFFFFF 2>&1 | FileCheck %s --check-prefixes=MAX,HINT-HIGHADDR
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0x0000000000000000 2>&1 | FileCheck %s --check-prefixes=ZERO,HINT-PAGE0
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0x0000000000000FFF 2>&1 | FileCheck %s --check-prefixes=LOW1,HINT-PAGE0
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0x0000000000001000 2>&1 | FileCheck %s --check-prefixes=LOW2,HINT-NONE
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0x4141414141414141 2>&1 | FileCheck %s --check-prefixes=HIGH,HINT-HIGHADDR
+// RUN: %env_asan_opts=print_scariness=1 not %run %t 0xFFFFFFFFFFFFFFFF 2>&1 | FileCheck %s --check-prefixes=MAX,HINT-HIGHADDR
```
This seems unrelated to this patch, is there a reason its included here?

connieyzhu replied:
I included it because it was related to the RUN line that caused the InternalShellError. I'm guessing I should have put this change in a separate patch?

ilovepi reviewed (changes requested):
I'm fine w/ the code change in TestRunner.py, but 

1) this needs a test in lit's test suite
1) the compiler-rt change should be split out on its own

connieyzhu committed:
[llvm-lit][test] Created test for built-in export with too many args

This patch creates a test to show the failing behavior for lit's
built-in export command when given more than one argument.

ilovepi reviewed (approved):
LGTM.
